### ISSUE TITLE
handleRedirectPromise throws an empty error for random users

### ISSUE THREAD

**Issue by roldengarm** [AzureAD/microsoft-authentication-library-for-js#6756]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

3.6.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

n/a

### Public or Confidential Client?

Public

### Description

For our Word add-in, we have a login component that handles the authentication with MSAL JS. It's a popup and it calls `handleRedirectPromise`. 
For a set of our users, an error is thrown, but the error object is empty.
Our code is as follows:

```
 this.msalInstance.handleRedirectPromise().then(() => // do something with response).catch(err => {
// handle error
// this is randomly an empty object
});
```
We've upgraded to the latest MSAL package but issue persists.

While the add-in is using Angular, we don't use MSAL Angular, just MSAL JS.

### Error Message

{}

### MSAL Configuration

```javascript
{
auth: {
      clientId: this.authConfiguration.clientId,
      authority: this.authConfiguration.authority,
      redirectUri: `${window.location.origin}/auth/login`,
      knownAuthorities: [B2C_AUTHORITY],
    },
    cache: {
      cacheLocation: 'localStorage', // needed to avoid "login required" error
      storeAuthStateInCookie: true, // recommended to avoid certain IE/Edge issues
    },
}
```

### Relevant Code Snippets

```javascript
See above
```

### Reproduction Steps

See above

### Expected Behavior

No empty errors occur.

### Identity Provider

Azure B2C Custom Policy

### Browsers Affected (Select all that apply)

Other

### Source

External (Customer)

**Comment by roldengarm** [comment]
Upon enabling trace logging, I can see this in the logs just before the issue happens:
```
@azure/msal-browser@3.6.0 : Info - Emitting event: msal:handleRedirectStart
@azure/msal-browser@3.6.0 : Verbose - handleRedirectPromise has been called for the first time, storing the promise
@azure/msal-browser@3.6.0 : Trace - BrowserCacheManager.getCachedNativeRequest called
@azure/msal-browser@3.6.0 : Trace - BrowserCacheManager.getTemporaryCache: No cache item found in local storage
@azure/msal-browser@3.6.0 : Trace - BrowserCacheManager.getCachedNativeRequest: No cached native request found
@azure/msal-browser@3.6.0 : Trace - handleRedirectPromise - acquiring token from web flow
@azure/msal-browser@3.6.0 : Trace -  BrowserCacheManager.getTemporaryCache: No cache item found in local  storage
```
Then, the next thing is that the `catch` method is called with an empty object (`{}`)

It seems to mainly impact Mac users. For most users it has worked before, and it suddenly stopped working

**Comment by roldengarm** [comment]
I've found out why the error was an empty object. This is because an `Error` object by default doesn't serialize properly.
After serializing it better, I could see the unexpected error that causes this issue. 
Please see below the value of 'message' of the error being thrown:

```
server_error: AADB2C90273: An invalid response was received : 'Error: server_error'\r\nCorrelation ID: e1c845a9-d0c2-4348-858e-81c004976bc8\r\nTimestamp: 2023-12-11 17:33:51Z\r\n","errorCode":"server_error","errorMessage":"AADB2C90273: An invalid response was received : 'Error: server_error'\r\nCorrelation ID: e1c845a9-d0c2-4348-858e-81c004976bc8\r\nTimestamp: 2023-12-11 17:33:51Z\r\n","subError":"","name":"ServerError","correlationId":"6f44051b-1803-4679-b2e6-e9eb8942859d"
```
There is zero information on this error code, the only thing I can find is other people with the same issue.

**Comment by roldengarm** [comment]
Update: Upon further investigation, discovered why the error is empty, because it's a standard `Error` object which doesn't serialize properly.
When serializing, I can see that this error is thrown: 

```
window.crypto.randomUUID is not a function. (In 'window.crypto.randomUUID()', 'window.crypto.randomUUID' is undefined
```
This call to randomUUID is in the MSAL package. See here: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/21afd757cd901dc1b2a0f6ce663ab05166e12067/lib/msal-browser/src/crypto/BrowserCrypto.ts#L87

Why is this happening?

**Comment by tnorling** [comment]
Please confirm [your browser supports randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) also note that crypto APIs are only supported in secure contexts.

**Comment by roldengarm** [comment]
@tnorling I've just added additional logging and it failed for one user on Safari 15.2; which I don't believe has randomUUID.

We're on Angular 16, which ships with browerslist for support including Safari 15. However, [I now see that MSAL supports](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/FAQ.md) the last 2 browsers, which does not include Safari 15.2.

As a quick fix I've added a polyfill for randomUUID. 

I'm not 100% sure this fixes it for everyone as we've had reports from people on Mac OS 12 as well which should have a newer browser version.
